posthog/models/event.py

```diff
@@ -20,67 +20,96 @@
 from .team import Team
 
 from posthog.tasks.slack import post_event_to_slack
-from typing import Dict, Union, List, Optional, Any
+from typing import Dict, Union, List, Optional, Any, Tuple
 
 import re
+import copy
 
 attribute_regex = r"([a-zA-Z]*)\[(.*)=[\'|\"](.*)[\'|\"]\]"
 
 
-def split_selector_into_parts(selector: str) -> List:
-    tags = selector.split(" > ")
-    tags.reverse()
-    ret: List[Dict[str, Union[str, List]]] = []
-    for tag in tags:
-        data: Dict[str, Union[str, List]] = {}
+class SelectorPart(object):
+    direct_descendant = False
+
+    def __init__(self, tag: str, direct_descendant: bool):
+        self.direct_descendant = direct_descendant
+        self.data: Dict[str, Union[str, List]] = {}
+
         result = re.search(attribute_regex, tag)
         if result and "[id=" in tag:
-            data["attr_id"] = result[3]
+            self.data["attr_id"] = result[3]
             tag = result[1]
         if result and "[" in tag:
-            data["attributes__{}".format(result[2])] = result[3]
+            self.data["attributes__{}".format(result[2])] = result[3]
             tag = result[1]
         if "nth-child(" in tag:
             parts = tag.split(":nth-child(")
-            data["nth_child"] = parts[1].replace(")", "")
+            self.data["nth_child"] = parts[1].replace(")", "")
             tag = parts[0]
         if "." in tag:
             parts = tag.split(".")
-            data["attr_class"] = parts[1:]
+            self.data["attr_class__contains"] = parts[1:]
             tag = parts[0]
         if tag:
-            data["tag_name"] = tag
-        ret.append(data)
-    return ret
+            self.data["tag_name"] = tag
+
+
+class Selector(object):
+    parts: List[SelectorPart] = []
+
+    def __init__(self, selector: str):
+        self.parts = []
+        tags = re.split(" ", selector)
+        tags.reverse()
+        for index, tag in enumerate(tags):
+            if tag == ">":
+                continue
+            direct_descendant = False
+            if tags[index - 1] == ">":
+                direct_descendant = True
```

Comment:
```
tags = ['div', '>', 'span', '>', 'a']
tags.reverse()
index = 0
tags[index - 1] == 'div'
```

Is that the right thing to check? It feels like it's checking the wrong thing and there might be some room for abuse here, though I couldn't come up with something that would make this crash... other than a `"> div span"` or  `"div span >"` type incorrect query potentially having the `direct_descendant` flag wrong.

Still might be worth pushing in an `if index > 0 and tags[...` clause?

Reply:
Added